opencensus/tags/tag_map.py

```diff
@@ -20,71 +25,69 @@ class TagMap(object):
     :param tags: a list of tags
 
     """
-    def __init__(self, tags=None):
-        self._map = {}
-        if tags is not None:
-            self.tags = tags
-            for tag in self.tags:
-                for tag_key, tag_value in tag.items():
-                    self._map[tag_key] = tag_value
 
-        else:
-            self._map = {}
+    def __init__(self, tags=None):
+        self.map = OrderedDict(tags if tags else [])
 
-    @property
-    def map(self):
-        """The current map of tags"""
-        return self._map
+    def __iter__(self):
+        return self.map.items().__iter__()
 
     def insert(self, key, value):
         """Inserts a key and value in the map if the map does not already
         contain the key.
 
-        :type key: :class: '~opencensus.tags.tag_key.TagKey'
+        :type key: Union[str, :class: '~opencensus.tags.tag_key.TagKey']
```

Comment:
I think it's better to keep `key` as `TagKey` and `value` as `TagValue`, to be consistent with other methods in this class.